redash/tasks/general.py

```diff
@@ -63,6 +65,33 @@ def send_mail(to, subject, html, text):
         logger.exception("Failed sending message: %s", message.subject)
 
 
+@job("queries", timeout=30, ttl=90)
+def test_connection(data_source_id):
+    try:
+        data_source = models.DataSource.get_by_id(data_source_id)
+        data_source.query_runner.test_connection()
+    except Exception as e:
+        return e
+    else:
+        return True
+
+
+@job("schemas", queue_class=Queue, at_front=True, timeout=300, ttl=90)
+def get_schema(data_source_id, refresh):
+    try:
+        data_source = models.DataSource.get_by_id(data_source_id)
+        return data_source.query_runner.get_schema(refresh)
```

Comment:
```suggestion
        return data_source.get_schema(refresh)
```

The data source method has additional logic (caching & sorting). The Query Runner is a more lower level and while it happens to accept a boolean parameter, it's not for refreshing (it's always refreshing).